### PR TITLE
Fix error when Bearer token is requested with empty string

### PR DESCRIPTION
**Pull request by jacknie84**
Issue gh-15885

### CODE REVIEW

**Review by sjohnr — CHANGES_REQUESTED** [review]
Hi @jacknie84, thanks for the PR! Please see comment below. Also there are failures on the build. Let me know if you need any assistance with the build output.

File: oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/web/DefaultBearerTokenResolverTests.java [hunk 1]
```diff
@@ -258,4 +261,36 @@ public void resolveWhenQueryParameterIsPresentAndNotSupportedThenTokenIsNotResol
 		assertThat(this.resolver.resolve(request)).isNull();
 	}
 
+	@Test
+	public void resolveWhenQueryParameterIsPresentAndEmptyStringThenTokenIsNotResolved() {
+		this.resolver.setAllowUriQueryParameter(true);
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.setMethod("GET");
+		request.setQueryString("access_token=");
+		request.addParameter("access_token", "");
```
[the remark this answers is not in the document]
Comment (jacknie84) [reply oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/web/DefaultBearerTokenResolverTests.java:270]: Yes, I'll remove it.

**Comment by jacknie84** [comment]
Hi @sjohnr. Thank you for your review.
I've made all the changes you asked for.
and I updated the test code because it was failing due to recent code changes.
Please review it once again.

File: oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/web/server/authentication/ServerBearerTokenAuthenticationConverterTests.java [hunk 2]
```diff
@@ -217,6 +217,19 @@ void resolveWhenQueryParameterHasMultipleAccessTokensThenOAuth2AuthenticationExc
 
 	}
 
+	@Test
+	public void resolveWhenQueryParameterIsPresentAndEmptyStringThenTokenIsNotResolved() {
```
Comment (sjohnr) [inline oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/web/server/authentication/ServerBearerTokenAuthenticationConverterTests.java:221]: I believe this is the same test as `resolveWhenQueryParameterIsEmptyAndSupportedThenOAuth2AuthenticationException()`? I think this test isn't needed?

Comment (jacknie84) [reply oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/web/server/authentication/ServerBearerTokenAuthenticationConverterTests.java]: Hi @sjohnr.
This test was deemed unnecessary and was removed. thank you.

**Comment by sjohnr** [comment]
@jacknie84 the PR Build failed due to code formatting issues related to imports. Let me know if you need any assistance.

**Comment by jacknie84** [comment]
@sjohnr Sorry~
The results of the ':spring-security-oauth2-resource-server:checkstyleTest' task were checked and action was taken.
Thank you.

**Review by sjohnr — APPROVED** [review]

**Comment by sjohnr** [comment]
Thanks @jacknie84! This has been merged into `6.2.x` as 4c6fef82b9b8d2a985305bcd1c0d59cc60245583.

**Pull request closed without merging**